src/pages/AppList/FiltersAndSorts.ts

```diff
@@ -140,12 +135,8 @@ export namespace AppListFilters {
     if (sortField.title === 'Health') {
       // In the case of health sorting, we may not have all health promises ready yet
       // So we need to get them all before actually sorting
-      const allHealthPromises: Promise<AppListItemHealth>[] = unsorted.map(item => {
-        return item.healthPromise.then(health => {
-          const withHealth: any = item;
-          withHealth.health = health;
-          return withHealth;
-        });
+      const allHealthPromises: Promise<WithAppHealth<AppListItem>>[] = unsorted.map(item => {
+        return item.healthPromise.then((health): WithAppHealth<AppListItem> => ({ ...item, ...{ health } }));
```

Comment:
Minor: `...{ health }` can be simplified to just `health` I think? @israel-hdez pointed it out in earlier review iirc.
Same thing occurs 3 times (app/workloads/service sorting)